source/header_cell_view/header_cell_view.tsx

```diff
@@ -1,18 +1,18 @@
 import * as React from 'react';
-import {Sorting} from '../sorting';
-import {FilterIconButton} from './filter_icon_button';
-import {SortIcon} from './sort_icon';
+import { SortOrder } from '../sort_order';
+import { FilterIconButton } from './filter_icon_button';
+import { SortIcon } from './sort_icon';
 
 interface Properties {
 
-  /** The full name of the header cell */
+  /** The full name of the header cell. */
   name: string;
 
-  /** The condensed name to used. */
+  /** The shortened name of the header cell. */
   shortName: string;
 
   /** The sort order of the column. */
-  sortOrder: Sorting;
+  sortOrder: SortOrder;
 
   /** Indicates if the sort button should be shown. */
```

Comment:
When would the sort button not be shown? The name also conflicts with ```sortOrder```.

Reply:
The sort indicator is only visible for the most recently sorted column.

A simple case would be Column A is sorted. Then column B is sorted. A is still sorted, but the sort indicator won't be shown.

Reply:
Look at eliminating one of ```sortOrder``` or ```isSorted```.

Reply:
I think you mean `isSortable` not `isSorted`. One indicates if a column can potentially be sorted and the other indicates if it's actually sorted. The functionality of `isSorted` is achieved by SortOrder already.

I do need an `isSortShown`.

Reply:
Sure go for it, to be honest I don't really know what the difference between them is. Two boolean values can produce 4 possible combinations and it's not clear to me what the interpretation of all those combinations are. Some of them make sense but a lot of them don't.